wp-content/plugins/core/src/Templates/Single.php

```diff
@@ -102,4 +106,41 @@ protected function get_pagination_item( $post_id ): string {
 		return $link->render();
 	}
 
+	protected function get_related_posts_cards() {
+		global $post;
```

Comment:
Hmm, is there a reason your calling global $post here and then overriding it in your foreach? Was WP doing the thing where certain post-related functions like get_the_content() weren't working without it?